### PR TITLE
feat(react-component): adding context menu per chart

### PR DESCRIPTION
**Pull request by mnischay**
## Overview
Adding context menu for charts. this will be per chart i.e when in dashboard this will not be shown instead the dashboard will be updated with the TC options in the follow up CR. 

## Verifying Changes
Refer to the screen capture

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

https://github.com/awslabs/iot-app-kit/assets/135036199/76a8f5e2-6daa-45b6-9fb6-e46372f96cd0

### CODE REVIEW

File: packages/react-components/package.json [hunk 1]
```diff
@@ -99,6 +99,7 @@
     "@iot-app-kit/core-util": "7.1.0",
     "@iot-app-kit/source-iottwinmaker": "7.1.0",
     "color": "^4.2.3",
+    "copy-to-clipboard": "^3.3.3",
```
Comment (tracy-french) [inline packages/react-components/package.json:102]: https://github.com/sudodoki/copy-to-clipboard uses a deprecated API and hasn't been updated in a year. Might want to consider https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Interact_with_the_clipboard#writing_to_the_clipboard.

File: packages/react-components/src/components/chart/contextMenu/ChartContextMenu.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    <ChartContextMenu
+      menuOptionClickHandler={menuOptionClickHandler}
+      onOutSideClickHandler={onOutSideClickHandler}
+      position={{ x: 100, y: 100 }}
+      trendCursors={[]}
+    />
+  );
+  it('renders', () => {
+    expect(component).not.toBeNull();
+  });
```
Comment (tracy-french) [inline packages/react-components/src/components/chart/contextMenu/ChartContextMenu.spec.tsx:19]: Nothing more we should test, such as the existence of each action? Where would we catch the issue if one of the context menu actions was disable on accident?

File: packages/react-components/src/components/chart/hooks/useTrendCursorsEvents.ts [hunk 3]
```diff
@@ -151,17 +172,46 @@ const useTrendCursorsEvents = ({
...
+  const copyTrendCursorData = (posX: number) => {
+    const timestampOfClick = calculateTimeStamp(posX, sizeRef.current.width, viewportRef.current);
+    const toBeCopiedGraphicIndex = calculateNearestTcIndex(graphicRef.current, timestampOfClick);
+    // using copy-to-clipboard library to copy in a Excel sheet pastable format
+    copy(formatCopyData(graphicRef.current[toBeCopiedGraphicIndex], seriesRef.current), { format: 'text/plain' });
+  };
+
+  // this handles the user interaction via context menu
+  const onContextMenuClickHandler = ({ action, posX }: { action: Action; posX: number }) => {
+    switch (action) {
```
Comment (tracy-french) [inline packages/react-components/src/components/chart/hooks/useTrendCursorsEvents.ts:201]: Without a default, what happens if a fourth action is added and this switch statement isn't updated?

**Review by tracy-french — APPROVED** [review]
Thank you @mnischay! Please consider my feedback in future iterations. :)

File: packages/react-components/src/components/chart/hooks/useTrendCursorsEvents.ts (hunk 3 above)
Comment (mnischay) [reply packages/react-components/src/components/chart/hooks/useTrendCursorsEvents.ts:201]: No action occurs, did not want to have any action if by chance someone changes the Action type

File: packages/react-components/src/components/chart/contextMenu/ChartContextMenu.spec.tsx (hunk 2 above)
Comment (mnischay) [reply packages/react-components/src/components/chart/contextMenu/ChartContextMenu.spec.tsx:19]: I am looking into how can we trigger events on charts, havent been successful so far.

File: packages/react-components/package.json (hunk 1 above)
Comment (mnischay) [reply packages/react-components/package.json:102]: given the number of usages / downloads, i assumed it to be ok to use... https://www.npmjs.com/package/copy-to-clipboard

**Pull request merged**